### PR TITLE
[SERVICE-239] Maximized Windows + Tabbing Improvements

### PR DESCRIPTION
**Pull request by ryanpetersonOF**
Fixes some maximize + tabbing logic

### CODE REVIEW

File: src/provider/tabbing/GroupWindow.ts [hunk 1]
```diff
@@ -123,8 +123,10 @@ export class GroupWindow extends AsyncWindow {
      */
     public async restoreGroup(): Promise<void|void[]> {
         if (this._isMaximized) {
-            if ((await this.getState()) === 'minimized') {
-                return this._tabGroup.activeTab.window.restore();
+            if (await this._tabGroup.activeTab.window.getState()) {
```
Comment (tomrobertsOF) [inline src/provider/tabbing/GroupWindow.ts]: Should this not still have the `=== minimised`, or is there some other logic hidden away that handles that?

File: src/provider/tabbing/TabService.ts [hunk 2]
```diff
@@ -164,18 +164,23 @@ export class TabService {
 
         const firstTab = tabsP.shift();
 
-        if (firstTab) {
-            const bounds = await firstTab.window.getWindowBounds();
-            tabsP.forEach(tab => tab.window.finWindow.setBounds(bounds.left, bounds.top, bounds.width, bounds.height));
-            tabsP[tabsP.length - 1].window.finWindow.bringToFront();
-            await group.addTab(firstTab, false);
+        if (!firstTab) {
+            return Promise.reject('Something is wrong...');
```
Comment (tomrobertsOF) [inline src/provider/tabbing/TabService.ts]: Can you make this a more useful error message.

Comment (ryanpetersonOF) [reply src/provider/tabbing/TabService.ts]: It is actually impossible to reach this.  We check if the if array has length right away.  Shift() returns undefined on a 0 length array

Comment (tomrobertsOF) [reply src/provider/tabbing/TabService.ts]: If it will never get called, why is it there?

**Pull request merged**